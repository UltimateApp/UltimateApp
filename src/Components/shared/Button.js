import React from 'react';
import { StyleSheet, Text, TouchableOpacity } from 'react-native';

import theme from '../../styles/theme.style';

const Button = props => {
  return (
    <TouchableOpacity style={styles.button} onPress={props.onPress} testID="button">
      <Text style={styles.text}>{props.text}</Text>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  button: {
    left: '10%',
    right: '10%',
    width: '80%',
    height: 40,
    borderRadius: 30,
    backgroundColor: theme.COLOR_PRIMARY,
    padding: 10,
  },
  text: {
    textAlign: 'center',
    color: theme.COLOR_PRIMARY_LIGHT,
    fontSize: theme.FONT_SIZE_MEDIUM,
  },
});

export default Button;
