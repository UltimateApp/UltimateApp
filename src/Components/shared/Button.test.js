import React from 'react';
import renderer from 'react-test-renderer';
import { render, fireEvent, cleanup } from 'react-native-testing-library';

import Button from './Button';

afterEach(cleanup);

describe('<Button />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Button text="OK" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('calls delegates onPress prop', async () => {
    const onPress = jest.fn();

    const { getByTestId } = render(<Button text="Easy" onPress={onPress} />);

    await fireEvent.press(getByTestId('button'));

    expect(onPress).toHaveBeenCalled();
  });
});
