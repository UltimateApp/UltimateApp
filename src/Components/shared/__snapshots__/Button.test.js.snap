// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders correctly 1`] = `
<View
  accessible={true}
  focusable={false}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#232b2b",
      "borderRadius": 30,
      "height": 40,
      "left": "10%",
      "opacity": 1,
      "padding": 10,
      "right": "10%",
      "width": "80%",
    }
  }
  testID="button"
>
  <Text
    style={
      Object {
        "color": "#fff",
        "fontSize": 16,
        "textAlign": "center",
      }
    }
  >
    OK
  </Text>
</View>
`;
